### PR TITLE
Add region label to ManagedCluster when available

### PR DESCRIPTION
**Pull request by jnpacker**
Signed-off-by: Joshua Packer <jpacker@redhat.com>

* Adds the region label to the created ManagedCluster when it is available. For the main cloud providers.
* Tests added as well to verify it works for AWS, GCP & Azure

### CODE REVIEW

**Comment by jnpacker** [comment]
Bug Reference: https://github.com/open-cluster-management/backlog/issues/16456

**Comment by jnpacker** [comment]
WIP because I need to wait for the Role merge in the cluster-lifecycle-chart (https://github.com/open-cluster-management/cluster-lifecycle-chart/pull/43), so that the new permissions needed to read the ClusterDeployment are available before we commit this change.

File: controllers/clusterclaims/clusterclaims-controller.go [hunk 1]
```diff
@@ -264,3 +269,39 @@ func deleteResources(r *ClusterClaimsReconciler, target string) error {
...
+	var cd hivev1.ClusterDeployment
+	log := r.Log
+
+	if err := r.Client.Get(
+		context.Background(),
+		types.NamespacedName{Namespace: clusterName,
+			Name: clusterName}, &cd); err != nil {
+
+		if k8serrors.IsNotFound(err) {
+			r.Log.V(WARN).Info("No ClusterDeployment found for " + clusterName)
```
Comment (ldpliu) [inline controllers/clusterclaims/clusterclaims-controller.go:285]: should return nil here, right ?

Comment (ldpliu) [reply controllers/clusterclaims/clusterclaims-controller.go:285]: It's ok for me. Thank you

**Review by ldpliu — APPROVED** [review]

**Pull request merged**